common-http/src/main/java/co/cask/common/http/HttpRequests.java

```diff
@@ -79,6 +79,9 @@ public static HttpResponse execute(HttpRequest request, HttpRequestConfig reques
     InputSupplier<? extends InputStream> bodySrc = request.getBody();
     if (bodySrc != null) {
       conn.setDoOutput(true);
+      if (request.hasBodyLength()) {
```

Comment:
Otherwise, use the chunk mode to avoid `HttpURLConnection` to cache anything at all.

```conn.setChunkedStreamingMode(0);  // to use default chunk size of 4096```